### ISSUE TITLE
🚀 Feature: Migration should warn user when they skip more than one version (or run twice).

### ISSUE THREAD

**Issue by gewenyu99** [appwrite/appwrite#3176]
### 🔖 Feature description

Migration script currently allows the user to break their deployment by upgrading too many major versions. 

For example, 0.11->0.12 is fine, 0.11->0.13 should not break the user's deployment.

The script should either not run, handle the situation gracefully, or error out when it's being used incorrectly.

TL;DR The migration script should not allow users to upgrade more than 1 major version at a time and break their deployments.

### 🎤 Pitch

As we release new versions more frequently and the product becomes more mature, it's important to safeguard users from accidentally upgrading more than one major version at a time and breaking their deployment. This is easily avoided by the migration script checking the current and target versions.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by gewenyu99** [comment]
@christyjacob4 I've assigned this to you, for now, so we don't forget about this. Please schedule this into a future release 😃

**Comment by stnguyen90** [comment]
[This](https://github.com/appwrite/appwrite/issues/2616#issuecomment-1641223048) should be implemented first and then this should be a separate PR on top.

**Comment by BadgerBloke** [comment]
> [This](https://github.com/appwrite/appwrite/issues/2616#issuecomment-1641223048) should be implemented first and then this should be a separate PR on top.

Hi @stnguyen90 I would like to work on this. can you please assign this to me

https://github.com/appwrite/appwrite/issues/2616 points to https://github.com/appwrite/appwrite/issues/3176

and here you are pointing to https://github.com/appwrite/appwrite/issues/2616

**Comment by btme0011** [comment]
can i work on this?

**Comment by stnguyen90** [comment]
This would be nice, but I'm not sure how the migrate command would check if the admin is skipping versions 🧐

**Comment by stnguyen90** [comment]
@gewenyu99, 

I'm unsure if it's possible to check versions when running migrate...How about giving some warning if they are upgrading and the new version has is more than 2 minor versions? Also what should the copy/warning text be?

You also mentioned something about running twice. Can you clarify what you meant?

**Comment by gewenyu99** [comment]
> @gewenyu99,
> 
> I'm unsure if it's possible to check versions when running migrate...How about giving some warning if they are upgrading and the new version has is more than 2 minor versions? Also what should the copy/warning text be?

Yeah, let's make them have to type Y/N to see if they're skipping versions.

> 
> You also mentioned something about running twice. Can you clarify what you meant?

If you try to run the migration script twice, i.e. ur on 1.3 and u run migrations again. This will still run the script, with undefined outcomes last I checked.

**Comment by BadgerBloke** [comment]
@stnguyen90 and @gewenyu99 since we don't have the current version info anywhere then, as you said @gewenyu99 to ask the user that are they skipping the version or not. For the time being it sounds good but it will be better if we start storing the version info in db.

**Comment by gewenyu99** [comment]
@BadgerBloke I agree. 😅 Seems like we just need to fix this at a product level and just make it idiot proof (I'd love to be able be an idiot)

**Comment by BadgerBloke** [comment]
> @BadgerBloke I agree. 😅 Seems like we just need to fix this at a product level and just make it idiot proof

Yes exactly


I have requested @stnguyen90 for assigning this to me

**Comment by DH-555** [comment]
This is def very needed
